### PR TITLE
Global styling

### PR DESCRIPTION
**Pull request by SabinePfanner**
(!) Ticket requested some fundamental changes within the card list - implemented Grid instead of Flexbox logic 

Otherwise, followed US: https://github.com/users/SabinePfanner/projects/2/views/1?pane=issue&itemId=66506912

### CODE REVIEW

File: components/StyledElements/Buttons.js [hunk 1]
```diff
@@ -3,15 +3,32 @@ import styled from "styled-components";
...
-  background-color: var(--color-green);
+  background-color: var(--primary-light);
   padding: 0.5rem;
   border-radius: 0.5rem;
   width: 5.5rem;
-  font-weight: bold;
-  border: 2px solid green;
+  font-weight: 700;
+  border: 2px solid var(--primary-dark);
+  color: var(--primary-contrast)
```
Comment (ChristianMoschke) [inline components/StyledElements/Buttons.js]: ';' expected

File: components/StyledElements/Buttons.js [hunk 2]
```diff
@@ -3,15 +3,32 @@ import styled from "styled-components";
...
+`;
+
+export const StyledButtonCancel = styled.button`
+  background-color: var(--secondary-light-200);
+  padding: 0.5rem;
+  border-radius: 0.5rem;
+  width: 5.5rem;
+  font-weight: 700;
+  border: 2px solid var(--secondary-light);
+  color: var(--primary-contrast)
```
Comment (ChristianMoschke) [inline components/StyledElements/Buttons.js]: ';' expected

File: components/CardList.js [hunk 3]
```diff
@@ -3,56 +3,43 @@ import styled from "styled-components";
...
-  max-width: 300px;
-  background-color: #fff;
-  border-radius: 8px;
-  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
-  overflow: hidden;
-  transition: transform 0.2s;
+const StyledList = styled.ul`
+  list-style: none;
+  display: grid;
+  grid-template-columns: repeat(4, 1fr);
```
Comment (ChristianMoschke) [inline components/CardList.js:21]: Property grid-template-columns is overwritten

**Comment by ChristianMoschke** [comment]
In `components/Header.js` in line 26 is an unused parameter `onOpenToast`

**Comment by ChristianMoschke** [comment]
In `components/Login.js` is an unused import `signOut`

**Comment by ChristianMoschke** [comment]
In `components/StyledElements/RadioButton.js` line 14 is a mismatched property value `width: 12x`.

File: components/StyledElements/SvgIcon.js [hunk 4]
```diff
@@ -35,6 +35,34 @@ export const icons = {
...
+  },
+
+  herb: {
+    title: "Herb",
+    viewBox: "0 0 24 24",
+    path: "M7.33,18.33C6.5,17.17 6.5,15.83 6.5,14.5C8.17,15.5 9.83,16.5 10.67,17.67L11,18.23V15.95C9.5,15.05 8.08,14.13 7.33,13.08C6.5,11.92 6.5,10.58 6.5,9.25C8.17,10.25 9.83,11.25 10.67,12.42L11,13V10.7C9.5,9.8 8.08,8.88 7.33,7.83C6.5,6.67 6.5,5.33 6.5,4C8.17,5 9.83,6 10.67,7.17C10.77,7.31 10.86,7.46 10.94,7.62C10.77,7 10.66,6.42 10.65,5.82C10.64,4.31 11.3,2.76 11.96,1.21C12.65,2.69 13.34,4.18 13.35,5.69C13.36,6.32 13.25,6.96 13.07,7.59C13.15,7.45 13.23,7.31 13.33,7.17C14.17,6 15.83,5 17.5,4C17.5,5.33 17.5,6.67 16.67,7.83C15.92,8.88 14.5,9.8 13,10.7V13L13.33,12.42C14.17,11.25 15.83,10.25 17.5,9.25C17.5,10.58 17.5,11.92 16.67,13.08C15.92,14.13 14.5,15.05 13,15.95V18.23L13.33,17.67C14.17,16.5 15.83,15.5 17.5,14.5C17.5,15.83 17.5,17.17 16.67,18.33C15.92,19.38 14.5,20.3 13,21.2V23H11V21.2C9.5,20.3 8.08,19.38 7.33,18.33Z",
+  },
+  other: {
+    title: "Other crop typle",
```
Comment (ChristianMoschke) [inline components/StyledElements/SvgIcon.js]: Literal error: `typle`

**Comment by ChristianMoschke** [comment]
There is a duplicate in `index.js` line 27 and `my-garden.js` line 41.

``` javascript
  function toggleFilter(category, option) {
    setFilter((prevFilters) => {
      // if it's in, remove
      const newCategoryFilters = prevFilters[category].includes(option)
        ? prevFilters[category].filter((item) => item !== option)
        : [...prevFilters[category], option];
      return { ...prevFilters, [category]: newCategoryFilters };
    });
  }

**Review by ChristianMoschke — CHANGES_REQUESTED** [review]
Great work so far!
I found a couple things you should take care of.

File: components/CardList.js (hunk 3 above)
Comment (SabinePfanner) [reply components/CardList.js:21]: I do not get this comment - what is the issue? ```Grid-template-columns```is overwritten with media-queries to adapt the grid based on different screen sizes - that is on purpose or did you mean something else?

**Comment by SabinePfanner** [comment]
> There is a duplicate in `index.js` line 27 and `my-garden.js` line 41.
> 
> ```js
>   function toggleFilter(category, option) {
>     setFilter((prevFilters) => {
>       // if it's in, remove
>       const newCategoryFilters = prevFilters[category].includes(option)
>         ? prevFilters[category].filter((item) => item !== option)
>         : [...prevFilters[category], option];
>       return { ...prevFilters, [category]: newCategoryFilters };
>     });
>   }
> ```

Yes, is is expected as we did not want to move the filter logic on app level to handle it independently on each page,

File: components/CardList.js (hunk 3 above)
Comment (ChristianMoschke) [reply components/CardList.js:21]:
``` javascript
  grid-template-columns: repeat(4, 1fr);
  grid-template-columns: auto;
```
Doesn't the second line overwrite the first one?

**Review by ChristianMoschke — APPROVED** [review]
LGTM!

**Pull request merged**